src/main/java/macrobase/analysis/classify/EWAppxPercentileOutlierClassifier.java

```diff
@@ -0,0 +1,104 @@
+package macrobase.analysis.classify;
+
+import macrobase.analysis.result.DatumWithNorm;
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.analysis.sample.ExponentiallyBiasedAChao;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+import macrobase.util.Periodic;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/*
+ Exponentially weighted approximate percentile-based streaming classifier
+ */
+public class EWAppxPercentileOutlierClassifier extends OutlierClassifier {
+    private ExponentiallyBiasedAChao<DatumWithNorm> reservoir;
+
+    private double currentThreshold = 0;
+
+    private final Periodic reservoirDecayer;
+    private final Periodic percentileUpdater;
+
+    List<OutlierClassificationResult> warmupOutput = new ArrayList<>();
+
+    public EWAppxPercentileOutlierClassifier(MacroBaseConf conf, Iterator<Datum> input) {
+        this(conf,
+             input,
+             conf.getInt(MacroBaseConf.SCORE_RESERVOIR_SIZE, MacroBaseDefaults.SCORE_RESERVOIR_SIZE),
```

Comment:
Can't think of anything off the top of my head. Hard to have a clean solutions since we can have arbitrary sequences of operators now.